### PR TITLE
api: rearange internal/api for growing number of sub-APIs

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- This is the first step in generalizing the existing `configapi` to work on arbitrary AttestationConfig data, not just AzureSEVSNP versions. In the process of doing so, some changes to the `internal/api` pkg are necessary to separate different subapis from each other more clearly.
- The main change is splitting fetcher & client into a generic and sub-api specific part.
- Rearrange the packages under `internal/api`. The generic fetcher and client are in `internal/api/fetcher` and `internal/api/client`, respectively. Sub-APIs are in `internal/api/versions` or `internal/api/attestationconfig`. Strictly speaking there should also be `internal/api/images` and `internal/api/cliinfo`, as those are also specified as separate APIs in our RFC folder. However, to keep the changes small I didn't change that. Using import aliases the naming can be made specific in the importing code (e.g. [here](https://github.com/edgelesssys/constellation/compare/feat/api/extend-configapi?expand=1#diff-06df0ccd1ff71379f21375eff055cdd48759f2d6b56366e06e18fb858940a835)). The naming scheme follows the recommendations from [1].

[1] https://go.dev/blog/package-names
### Additional info
- Testing:
  - created [separate branch](https://github.com/edgelesssys/constellation/commit/0384c84874354f8d8b4d8798d1cee321e8cc36c6) with same changes, [build-os-images](https://github.com/edgelesssys/constellation/actions/runs/5142594410) :green_circle: on that branch (test add+list of versionsapi cli), delete branch after successful run (test rm of versionsapi cli). [Purge run](https://github.com/edgelesssys/constellation/actions/runs/5143590805) :green_circle:.
- Upcoming changes will be: 
  - Rename `internal/api` to `internal/resourceapi`. We settled on the term `Constellation Resource API` to refer to the entirety of our APIs.
  - Generalize `AzureSEVSNPVersionGet` and `AzureSEVSNPVersionList` to work for arbitrary AttestationCfg data. To do so we will have to move the types to the config package. To access the different attestation type structs for unmarshalling. 

### Checklist

- [ ] Add labels (e.g., for changelog category)
- [ ] Link to Milestone

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -98,6 +98,7 @@ require (
 	github.com/spf13/afero v1.9.5
 	github.com/spf13/cobra v1.7.0
 	github.com/stretchr/testify v1.8.3
+	github.com/tj/assert v0.0.0-20171129193455-018094318fb0
```
Comment (malt3) [inline go.mod:101]:
```suggestion
```

This got in by accident! Also remove any usage of this in our code!

File: internal/api/attestationconfig/fetcher/fetcher_test.go [hunk 2]
```diff
@@ -14,19 +14,20 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/edgelesssys/constellation/v2/internal/api/configapi"
-	"github.com/stretchr/testify/assert"
+	"github.com/edgelesssys/constellation/v2/internal/api/attestationconfig"
 	"github.com/stretchr/testify/require"
+	"github.com/tj/assert"
```
Comment (malt3) [inline internal/api/attestationconfig/fetcher/fetcher_test.go:19]:
```suggestion
	"github.com/stretchr/testify/assert"
```

**Review by malt3 — APPROVED** [review]
very clean. Good work @derpsteb and @elchead!

**Comment by derpsteb** [comment]
After talking to Paul again I was conviced that the package name `versions` for `internal/api/versions` is less than ideal. It makes an import alias necessary, pretty much all the time. 

Will rename to `versionsapi` / `attestationconfigapi` respectively in the next PR.

File: internal/api/attestationconfig/attestation.go [hunk 3]
```diff
@@ -3,7 +3,14 @@ Copyright (c) Edgeless Systems GmbH
 
 SPDX-License-Identifier: AGPL-3.0-only
 */
-package configapi
+
+/*
+- AttesationCfg objects within API need to hold some version information (for sorting, recognizing latest)
+- Thus, existing config types (AWSNitroTPM, AzureSEVSNP, ...) can not be extended to implement apiObject interface
+- Instead, we need a separate type that wraps _all_ attestation types. In the codebase this is done using the AttestationCfg interface
+- The new type AttestationCfgGet needs to be located inside internal/config in order to implement UnmarshalJSON.
```
[the remark this answers is not in the document]
Comment (derpsteb) [reply internal/api/attestationconfig/attestation.go:11]: Ah. Right. This was just meant as a reminder for us. But I guess it is useful nonetheless.

File: internal/api/client/client.go [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+	s3Client                     *s3.Client
+	uploadClient                 *s3manager.Uploader
+	bucket                       string
+	distributionID               string
+	cacheInvalidationWaitTimeout time.Duration
+
+	dirtyPaths []string // written paths to be invalidated
+	DryRun     bool     // no write operations are performed
+
+	Log *logger.Logger
```
[the remark this answers is not in the document]
Comment (derpsteb) [reply internal/api/client/client.go:61]: yes. versionsapi/client makes use of them. Originally the code was in the same pkg and the variables were private. We wanted to keep changes to the client minimal for now as Malte is also working on changing this client right now. This seemed like the simplest solution. Will try to clean this up in a future PR.

File: internal/api/fetcher/fetcher.go
[the remark this answers is not in the document]
Comment (derpsteb) [reply internal/api/fetcher/fetcher.go:1]: We can discuss this, but I would not want to do this in this PR. From moving the code around it feels like the specialized fetchers are the point where stubbing should happen, but maybe there are better places.

**Pull request merged**